src/System.IO.Pipelines/Pipe.cs

```diff
@@ -293,15 +293,16 @@ internal void Commit()
                 // Always move the commit head to the write head
                 _commitHead = _writingHead;
                 _commitHeadIndex = _writingHead.End;
+            }
 
-                _length += _currentWriteLength;
-                // Do not reset if reader is complete
-                if (_maximumSizeHigh > 0 &&
-                    _length >= _maximumSizeHigh &&
-                    !_readerCompletion.IsCompleted)
-                {
-                    Reset(ref _writerCallback);
-                }
+            var currentLenght = Interlocked.Add(ref _length, _currentWriteLength);
```

Comment:
Considering that _length is always updated immediately after or immediately prior to acquiring the _sync lock, is introducing a new memory barrier worth it? Is this measurably better than what we had before?